src/cargo/util/context/mod.rs

```diff
@@ -523,6 +526,13 @@ impl GlobalContext {
             .borrow_mut()
     }
 
+    /// Local overlay configuration.
+    ///
+    /// See [`crate::sources::overlay::OverlayConfusionAttack`] for why you shouldn't use this.
+    pub fn local_overlays(&self) -> RefMut<'_, HashMap<SourceId, PathBuf>> {
```

Comment:
Should probably mention that the paths need to point to local registries, not just to any crate file or directory.